File: ParteJava/ProyectoPro3/ProyectoPro3Modelo/src/main/java/pe/edu/pucp/proyectopro3/modelo/notificaciones/Notificacion.java
package pe.edu.pucp.proyectopro3.modelo.notificaciones;

import java.util.Date;

public class Notificacion {
    private int idNotificacion;
    private String mensaje;
    private String tipoNotificacion;
    private Date fechaEnvio;
    private boolean leido;
    private int idUsuario;
    private TipoEvento tipEvento;

    public Notificacion(int idNotificacion, String mensaje, String tipoNotificacion, Date fechaEnvio, boolean leido, int idUsuario, TipoEvento tipEvento) {
        this.idNotificacion = idNotificacion;
        this.mensaje = mensaje;
        this.tipoNotificacion = tipoNotificacion;
        this.fechaEnvio = fechaEnvio;
        this.leido = leido;
        this.idUsuario = idUsuario;
        this.tipEvento = tipEvento;
    }

    public Notificacion() {

    }

    public int getIdNotificacion() {
        return idNotificacion;
    }

    public void setIdNotificacion(int idNotificacion) {
        this.idNotificacion = idNotificacion;
    }

    public String getMensaje() {
        return mensaje;
    }

    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }

    public String getTipoNotificacion() {
        return tipoNotificacion;
    }

    public void setTipoNotificacion(String tipoNotificacion) {
        this.tipoNotificacion = tipoNotificacion;
    }

    public Date getFechaEnvio() {
        return fechaEnvio;
    }

    public void setFechaEnvio(Date fechaEnvio) {
        this.fechaEnvio = fechaEnvio;
    }

    public boolean isLeido() {
        return leido;
    }

    public void setLeido(boolean leido) {
        this.leido = leido;
    }

    public TipoEvento getTipoEvento() {
        return tipEvento;
    }

    public void setTipoEvento(TipoEvento tipEvento) {
        this.tipEvento = tipEvento;
    }

    public int getIdUsuario() { return idUsuario; }

    public void setIdUsuario(int idUsuario) { this.idUsuario = idUsuario; }
}

File: ParteC#/ProyectoProgramacion3/ProyectoProgramacion3Model/Model/notificaciones/Notificacion.cs
namespace ProyectoProgramacion3Model.Model.notificaciones;
public class Notificacion{
    public int idNotificacion {  get; set; }
    public string mensaje { get; set; } = string.Empty;
    public string tipoNotificacion { get; set; } = string.Empty;
    public DateOnly fechaEnvio { get; set; }
    public int idUsuario { get; set; }
    public Boolean leido { get; set; }
    public TipoEvento tipoEvento { get; set; }
    public Notificacion(int idNotificacion, string mensaje, string tipoNotificacion, DateOnly fechaEnvio, int idUsuario, Boolean leido, TipoEvento tipoEvento)
    {
        this.idNotificacion = idNotificacion;
        this.mensaje = mensaje;
        this.tipoNotificacion = tipoNotificacion;
        this.fechaEnvio = fechaEnvio;
        this.idUsuario = idUsuario;
        this.leido = leido;
        this.tipoEvento = tipoEvento;
    }
    public Notificacion()
    {
    }
}